website/docs/getting-started/template-walkthrough.md

```diff
@@ -0,0 +1,142 @@
+
+:::note
+
+An `App` may contain many stacks, but that is outside the scope of this document.
+You can read more about [`App`][cdk-app] and [`Stack`][cdk-stack] in the CDK documentation.
+
+:::
+
+Once we have an instance of `stack`, we're ready to attach our first `Function`:
+
+```typescript
+const sayFunction = new Function(
+  stack,
+  "SayFunction",
+  async (event: { message: string }) => {
+    // This code will run on AWS Lambda
+    console.log(event.message);
+    return;
+  }
+);
+```
+
+This might not be the most interesting code in the world, but it offers us our first glimpse at the magic of Functionless.
+
+If you've used CDK before, you're probably familiar with the seperation it imposes between infrastructure and application code.
+Functionless bridges this gap by letting you write application code inline with the infrastructure it sits atop.
+
+This opens up a world of new possibilities for how code and infrastructure can interact.
+We can see this in action with the `StepFunction` Construct definition:
+
+```typescript
+new StepFunction(stack, "Workflow", async (event: { name: string }) => {
+  // This will run in an AWS Step Function
+  await sayFunction({ message: `Hello ${event.name}` });
+});
+```
+
+There is a lot to unpack in these three lines.
+
+First off, we're able to call the `sayFunction` Construct we just defined as if it were a standard nodejs `function`.
+The types inferred from arrow function in the `Function` definition above are preserved, giving us type-checking when we call `sayFunction`.
+In Functionless, types are maintained across service boundaries, removing a whole class of user error.
+
+Secondly, Functionless introspects how your runtime code interacts with its infrastructure and uses this information to automatically configure least-privilege policies.
+
+:::note
+
+CDK happens to have equivalent functionality for inferring IAM permissions for Step Functions, but this behavior does not extend to other runtime usages such as Lambda.
+
+In Functionless, this ability extends to any runtime usage as we'll see later when we [modify the example](./change-the-code).
+
+:::
+
+Lastly, if you've ever created an AWS Step Function before, you might be wondering where all the Amazon States Language (ASL) went.
+
+Functionless automatically transpiles the code you supply in the `StepFunction` arrow function into ASL.
+This gives you the benefits of a world-class managed workflow service without requiring you to give up the productivity of TypeScript.
+
+:::caution
+
+The target service-specific languages Functionless transpiles TypeScript into don't always have parity with TypeScript.
+As a result, some TypeScript syntax cannot be transpiled into different targets.
+
+To address this **Functionless publishes a TypeScript language server plugin** which is configured in the generated template to catch these cases.
+
+:::
```

Comment:
Can we write this from the positive?

"Functionless' goal is to represent the Typescript/ECMA contract in the target service-specific language (ASL in this case).
 
When we cannot correctly represent the ECMA contract, we either throw a friendly [error](link to error codes) or have [caveats](link to step functions caveats). Use the Functionless Typescript Language service plugin provided with the template to get realtime feedback while developing or see [setup plugin](a link)."

Reply:
Or perhaps save this limitation to an equivalent getting started/tutorial that is dedicated to Step Functions? We just need to tease the user into exploring more, we don't need to give them everything. I think limitations and how to deal with them can be a whole section.

Reply:
Removed this section can cover it somewhere else in the docs.